ceramic-kubo-rpc/src/http/dag.rs

```diff
@@ -0,0 +1,315 @@
+use std::{io::Cursor, str::FromStr};
+
+use actix_multipart::Multipart;
+use actix_web::{http::header::ContentType, web, HttpResponse, Scope};
+use anyhow::anyhow;
+use futures_util::StreamExt;
+use iroh_api::IpfsPath;
+use libipld::{cbor::DagCborCodec, ipld, json::DagJsonCodec, prelude::Encode};
+use serde::Deserialize;
+
+use crate::{dag, error::Error, http::AppState, IpfsDep};
+pub fn scope<T>() -> Scope
+where
+    T: IpfsDep + 'static,
+{
+    web::scope("/dag")
+        .service(web::resource("/get").route(web::post().to(dag_get::<T>)))
+        .service(web::resource("/put").route(web::post().to(dag_put::<T>)))
+        .service(web::resource("/resolve").route(web::post().to(resolve::<T>)))
+}
+
+const DAG_CBOR: &str = "dag-cbor";
+const DAG_JSON: &str = "dag-json";
+
+fn dag_cbor() -> String {
+    DAG_CBOR.to_string()
+}
+// used to provide default to query structs
+fn dag_json() -> String {
+    DAG_JSON.to_string()
+}
+
+#[derive(Debug, Deserialize)]
+struct GetQuery {
+    arg: String,
+    #[serde(rename = "output-codec", default = "dag_json")]
+    output_codec: String,
+}
+
+#[tracing::instrument(skip(data))]
+async fn dag_get<T>(
+    data: web::Data<AppState<T>>,
+    query: web::Query<GetQuery>,
+) -> Result<HttpResponse, Error>
+where
+    T: IpfsDep,
+{
+    let ipfs_path = IpfsPath::from_str(query.arg.as_str()).map_err(|e| Error::Invalid(e.into()))?;
+    match query.output_codec.as_str() {
+        DAG_JSON => Ok(HttpResponse::Ok()
+            .content_type(ContentType::json())
+            .body(dag::get(data.api.clone(), &ipfs_path, DagJsonCodec).await?)),
+        DAG_CBOR => Ok(HttpResponse::Ok()
+            .content_type(ContentType::octet_stream())
+            .body(dag::get(data.api.clone(), &ipfs_path, DagCborCodec).await?)),
+        _ => Err(Error::Invalid(anyhow!(
```

Comment:
Now that you have support for dag-jose, should there be support for it here too?

Reply:
Yeah lets do these as subsequent PRs